cmd/verify/verify.go

```diff
@@ -0,0 +1,216 @@
+	command := &cobra.Command{
+		Use:               "verify",
+		Short:             "Verify if arlonctl can be run",
+		Long:              "Verify if required kubectl,argocd,git access is present before profiles and bundles are created",
+		DisableAutoGenTag: true,
+		Example:           "arlonctl verify",
+		RunE: func(c *cobra.Command, args []string) error {
+			// Verify kubectl status
+			kubectlStatus, err := verifyKubectl()
+			if err != nil {
+				fmt.Println("Error while verifying kubectl status: ", err)
+			} else {
+				fmt.Println("Successfully verified kubectl status")
+			}
+
+			// Verify argocd status
+			argoStatus, err1 := verifyArgoCD()
+			if err1 != nil {
+				fmt.Println("Error while verifying argocd status: ", err1)
+			} else {
+				fmt.Println("Successfully verified argocd status")
+			}
+
+			// Verify git status
+			gitStatus, err2 := verifyGit()
+			if err2 != nil {
+				fmt.Println("Error while verifying git status: ", err2)
+			} else {
+				fmt.Println("Successfully verified git status")
+			}
+
+			// Verify capi status
+			capiStatus, err4 := verifyCapi()
+			if err4 != nil {
+				fmt.Println("Error while verifying capi status: ", err2)
+			} else {
+				fmt.Println("Successfully verified capi status")
+			}
+
+			// Verify arlon status
+			arlonStatus, err3 := verifyArlon()
+			if err3 != nil {
+				fmt.Println("Error while verifying arlon status: ", err2)
+			} else {
+				fmt.Println("Successfully verified arlon status")
+			}
+
+			fmt.Println()
+			if kubectlStatus && argoStatus && gitStatus && arlonStatus && capiStatus {
+				fmt.Println("All requirements are installed")
+			} else {
+				fmt.Println("The check for Arlon prerequisites failed. Please install the missing tool(s).")
+			}
+
+			return nil
+		},
+	}
+	return command
+}
+
+// Check if kubectl is installed and the kubeconfig is pointing to the correct kubeconfig
+func verifyKubectl() (bool, error) {
+	_, err := exec.LookPath("kubectl")
+	if err != nil {
+		return false, ErrKubectlInstall
+	}
+
+	//Check if kubeconfig is correct and kubectl commands are functional
+	_, errClusterInfo := exec.Command("kubectl", "cluster-info").Output()
+	if errClusterInfo != nil {
+		return false, ErrClusterInfo
+	}
+
+	return true, nil
+}
+
+// Check if argocd cli is installed and the account has admin access
+func verifyArgoCD() (bool, error) {
+	// Check if argocd is installed
+	_, err := exec.LookPath("argocd")
+	if err != nil {
+		return false, ErrArgoCD
+	}
+
+	//Check if argocd has access and auth-token is not expired
+	_, errAcc := exec.Command("argocd", "account", "list").Output()
+	if errAcc != nil {
+		return false, ErrArgoCDAuthToken
+	}
+	return true, nil
+}
+
+// Check if git cli is installed
+func verifyGit() (bool, error) {
+	_, err := exec.LookPath("git")
+	if err != nil {
+		return false, ErrGit
+	}
+	return true, nil
+}
+
+// Verify if the arlon services are running
+func verifyArlon() (bool, error) {
+	ns := exec.Command("kubectl", "get", "ns")
```

Comment:
It would be more elegant to get a kube client, and use the Kubernetes API to query the `arlon` namespace resource to see if it exists, as opposed to running `kubectl` on the command line and grepping its output.

Reply:
Sure I'll do that. I agree, this looks a bit unrefined